ivadomed/loader/utils.py

```diff
@@ -501,3 +503,81 @@ def reorient_image(arr, slice_axis, nib_ref, nib_ref_canonical):
     trans_orient = nib.orientations.ornt_transform(ras_orientation, ref_orientation)
     # apply transformation
     return nib.orientations.apply_orientation(arr_ras, trans_orient)
+
+def create_bids_dataframe(loader_params, derivatives):
+    """Create a dataframe containing all BIDS image files in a bids_path and their metadata.
+
+    Args:
+        loader_params (dict): Loader parameters, see :doc:`configuration_file` for more details.
+        derivatives (bool): If True, derivatives are indexed.
```

Comment:
This is great :100: ! It makes it flexible for the dataset where we don't have derivatives. Concerning your question, I think we should index the derivatives if this option is set to True, it will be easier for the loading if everything is in the df. This function should also verify availability (in my opinion), so if we want to index derivatives, only the subjects with the right target(s) and contrast(s) should be in the final df.

Reply:
Ok thanks, I also think it's better to check availability before the split, I'll look into that.

Reply:
Filtering based on derivatives availability is done in b3c9d34. We now have to determine what is the behavior if no matching derivatives are found (currently raise an error and exit).